src/main/java/frc/robot/auto/AutoRoutines.java

```diff
@@ -0,0 +1,135 @@
+                List.of((midPosition)),
+                // End 3 meters straight ahead of where we started, facing forward
+                new Pose2d(endingPosition, new Rotation2d(0)),
+                config);
+
+        var thetaController = new ProfiledPIDController(
+                AutoConstants.kPThetaController, AutoConstants.kIThetaController, AutoConstants.kDThetaController,
+                AutoConstants.kThetaControllerConstraints);
+        thetaController.enableContinuousInput(-Math.PI, Math.PI);
+
+        SwerveControllerCommand swerveControllerCommand = new SwerveControllerCommand(
+                exampleTrajectory,
+                m_robotDrive::getPose, // Functional interface to feed supplier
+                DriveConstants.kDriveKinematics,
+
+                // Position controllers
+                new PIDController(AutoConstants.kPXController, 0, 0),
+                new PIDController(AutoConstants.kPYController, 0, 0),
+                thetaController,
+                m_robotDrive::setModuleStates,
+                m_robotDrive);
+
+        Command autonomousLogCommand = new RunCommand(
+                () -> {
+                    Logger.getInstance().recordOutput("DriveSubsystem/ThetaControllerGoalPosition",
+                            thetaController.getGoal().position);
+                    Logger.getInstance().recordOutput("DriveSubsystem/ThetaControllerGoalVelocity",
+                            thetaController.getGoal().velocity);
+                    Logger.getInstance().recordOutput("DriveSubsystem/ThetaControllerSetpointPosition",
+                            thetaController.getSetpoint().position);
+                    Logger.getInstance().recordOutput("DriveSubsystem/ThetaControllerSetpointVelocity",
+                            thetaController.getSetpoint().velocity);
+                    Logger.getInstance().recordOutput("DriveSubsystem/ThetaControllerPositionError",
+                            thetaController.getPositionError());
+                    Logger.getInstance().recordOutput("DriveSubsystem/ThetaControllerVelocityError",
+                            thetaController.getVelocityError());
+                });
+return swerveControllerCommand;
+    }
+
+    public SequentialCommandGroup TwoBallAutoRoutine(Pose2d zeroPose) {
+        return new SequentialCommandGroup(
+                new InstantCommand(() -> m_robotDrive.resetOdometry(zeroPose)),
+                new InstantCommand(m_intake::intakeIn), //actually out lol 
+            
+                new RunCommand(() -> this.DriveBetweenPoints(zeroPose.getTranslation(),
+                 FieldPositions.R3, FieldPositions.R3, m_robotDrive), m_robotDrive).withTimeout(0.5),
+                 //turn on launcher have enough time to speed up 
+            
+                 new RunCommand(() -> m_launcher.spinFromDistance(Constants.LauncherConstants.heightOfHighHubReflectors/(Math.tan(m_limelight.getVerticalAngle()))), m_launcher),
```

Comment:
@Ugasisra0129 @john-j-oneill This `RunCommand` has no timeout, so it will never finish, causing the remaining commands in the group to never execute. You probably want to keep spinning the launcher while the other commands are running, so you should nest a `ParallelCommandGroup` here which has the launcher `RunCommand` as one argument and another `SequentialCommandGroup` as another

Reply:
Yep! That’s exactly what we wanted to do, but we weren’t sure how. Thanks!